platforms/ios/HelloCordova/Plugins/WizAssetsPlugin/WizAssetsPlugin.m

```diff
@@ -37,18 +51,21 @@ - (void)backgroundDownload:(CDVInvokedUrlCommand *)command {
 
         WizLog(@"downloading ---------------- >%@", url);
 
-        NSData *urlData = [NSData dataWithContentsOfURL:url];
-
-        if (urlData) {
-            // path to library caches
-            NSArray  *paths = NSSearchPathForDirectoriesInDomains(NSCachesDirectory, NSUserDomainMask, YES);          
-            NSString *documentsDirectory = [paths objectAtIndex:0]; 
-            NSString *gameDir = [[[NSBundle mainBundle] infoDictionary]   objectForKey:@"CFBundleName"];
-            NSString *fullDir = [NSString stringWithFormat:@"%@/%@/%@", documentsDirectory, gameDir, storePath];
-            NSString *filePath = [fullDir stringByAppendingPathComponent:fileName];
-
-            if ([filemgr createDirectoryAtPath:fullDir withIntermediateDirectories:YES attributes:nil error: NULL] == YES) {
-
+        NSError *error = nil;
+        NSData *urlData = [NSData dataWithContentsOfURL:url options:NSDataReadingUncached error:&error];
+
+        if (error) {
+            returnString = [NSString stringWithFormat:@"error - %@", error];
+        } else if (urlData) {
+            // Check if we didn't received a 401
+            // TODO: We might want to find another solution to check for this kind of error, and check other possible errors
```

Comment:
We sure do. :)
```NSData``` is not the right way to download files in our case so it will be changed to use ```NSURLSession``` in a next PR.